### ISSUE TITLE
Update to latest stable Angular

### ISSUE THREAD

**Issue by jgw96** [driftyco/ionic#10422]
**Ionic version:**  (check one with "x")
[ ] **1.x**
[x] **2.x**

**I'm submitting a ...**  (check one with "x")
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or http://ionicworldwide.herokuapp.com/

Update to the latest stable angular release 2.4.7

**Comment by jgw96** [comment]
As per a discussion with brandy/dan today this issue is on hold while dan fixes an issue in app-scripts 1.1.0, which is required for this angular update.

**Comment by mburger81** [comment]
Can we know which bug do you talk about for app-scripts 1.1.0 and when do you think you will update to latest angular?

**Comment by AmitMY** [comment]
Also, just want to know if ionic is planned to support angular next big release (4) on march 1st, or not planned yet.

**Comment by jgw96** [comment]
@mburger81 Not blocked anymore, app-scripts 1.1.1 fixed the issue 😃  . Also @AmitMY , the plan is to support Angular 4 if they have it out by march 1st. Remember, their plan to have it out on march 1st is just the plan haha, things always change in software. We do plan to support it asap though once it comes out.

**Comment by jgw96** [comment]
Oh and i guess i should mention that you dont have to rush to update to app-scripts 1.1.1 as the changes were all related to e2e test type stuff for internal use. Feel free to update, but there are not many user facing changes

**Comment by mburger81** [comment]
good news to hear :+1:

**Comment by mburger81** [comment]
@jgw96 on update to app-scripts 1.1.1 http.get and post doesn't work anymore. Any http request gives error **404 Not Found**

**Comment by jgw96** [comment]
Status report on this: dan made some fixes to the e2e tests to get them to build correctly, currently waiting on his snapshot to finish running before we merge this update.

**Comment by jgw96** [comment]
Closing as the above PR was just checked off and merged.

**Comment by brandyscarney** [comment]
Closed by https://github.com/driftyco/ionic/commit/555519da620b3db3c7b0111a45dbb747e58e4f51

**Comment by carstenbaumhoegger** [comment]
this issue isn't really closed because of https://github.com/driftyco/ionic/commit/98e8158b9c03e142a50e79b6e754f9f0fa2a0233

**Comment by NgYueHong** [comment]
Yeah, looking forward for Angular 4 support as now Angular 4 is already in RC.

**Comment by jgw96** [comment]
Trying this again (: @NgYueHong we are steadily working on updating to Angular 4 (and very excited about it! ) but we most likely will not update to it until it is out of RC releases.

**Comment by jgw96** [comment]
Fixed with https://github.com/driftyco/ionic/pull/10592

**Comment by mikezks** [comment]
ad https://github.com/driftyco/ionic/pull/10592#issuecomment-284063919 by @mhartington 
>@mikezks Please re-read my comment, that is not what I said as all.

I could not answer your comment directly, becaue I was locked out of https://github.com/driftyco/ionic/pull/10592. It would be polite and show respect if you either at least ignore my post or give me the chance to reply to your comment before locking me out. Of course it is your project and therefore your decision. Furthermore expect that I read your comment - otherwise I would not answer. In case I got anything wrong a comment like "please re-read" is not very helpful in most of the cases - a comment on what I may have understood wrong can help.

Related to the context:
The main point I referred to was:
>since angular 2.2.1 there have been breaking changes that framework/library authors need take into consideration and update

IMO it does not make a big difference if someone uses angular as base for a framework or any other project. If these mentioned breaking changes within the angular framework affect someone's work it is not helpful if the angular team does not increment the most left version number. If these breaking changes were not intended, then there is less to discuss. 
I agree, in case you wanted to say something else, then I still have not understood your comment correctly.

Beside all above, thanks for all your work on the ionic framework. It is really amazing what you and your colleagues have achieved.

**Comment by mhartington** [comment]
Hey @mikezks was not trying to be rude or disrespectful, but just prevent constant back-and-forth on a closed PR. Not really the place for that.

>If these mentioned breaking changes within the angular framework affect someone's work it is not helpful if the angular team does not increment the most left version number.

Also, not trying to throw angular under the bus, but what we're seeing contradicts this. While the changes are not noted as "breaking", they do break bits of ionic or, for example ionic storage. Ionic storage was not setup correctly for angular 2.4.8 and if devs tried to update independently but got error, it's a bad user experience.

Locking the packages keeps things predictable. We know that X version of Ionic works with Y version angular.

**Comment by rapropos** [comment]
> Locking the packages keeps things predictable. We know that X version of Ionic works with Y version angular.

From the perspective of Ionic in a vacuum, this makes perfect sense. Once people start pulling in other dependencies to create apps, shared dependency mismatches are going to happen. It would be great if there was some publicly visible tracker that shows users what specifically is known to be incompatible with version >= x.y.z of Ionic's dependencies. Obviously, it's not reasonable to expect it to be exhaustive, but if we know what you know at least in terms of  land mines that have been hit, we can maybe help fix them and/or identify others. We can also make more informed decisions about whether our apps would be affected.

I realize you don't necessarily want to *encourage* independent dependency upgrading, and I respect that. A bit of information sharing that might make it somewhat less of a blind leap into the unknown would be a big help to app developers.

**Comment by mikezks** [comment]
Hi @mhartington, thanks for these further explanations - this helps to understand why you chose to pin the angular version.

I was thinking whether I may update the angular version in combination with ionic on my own risk, but this definitely sounds to be not a good idea ATM.

**Comment by sinedied** [comment]
@mhartington 
> Locking the packages keeps things predictable. We know that X version of Ionic works with Y version angular.

I understand issue may have happened during the Angular beta phase, but since then they fully respect [Semantic Versioning](http://semver.org), so allowing minor updates (or at least bufixes) can only improve the developer experience.

It's actually a pain in the *ss trying to maintain an Angular web app using Ionic (and a lot more other components) due to the Ionic's fixed dependencies.

If you really want to to maintain fixed dependencies for Ionic-centric projects, why not publish it as a separate npm package, like `ionic-angular-fixed`? This way everyone can be happy.
